apps/voting/app/src/components/Vote.js

```diff
@@ -0,0 +1,274 @@
+import { formatDate } from '../utils'
+
+const DEFAULT_DESCRIPTION =
+  'No additional description has been provided for this proposal.'
+const SEPARATOR = "for '"
+const EXECUTING = ' , executing'
+
+function Vote({ vote, onVote, onExecute }) {
+  const theme = useTheme()
+  const { layoutName } = useLayout()
+  const { tokenSymbol } = useAppState()
+  const { data, numData, voteId, connectedAccountVote } = vote
+  const { minAcceptQuorum, supportRequired, yea, nay } = numData
+  const { creator, endDate, open, metadata, description } = data
+  const text = description || metadata
+  const [rawTitle, extra = DEFAULT_DESCRIPTION] =
+    text.indexOf(SEPARATOR) > -1 ? text.split(SEPARATOR) : [text]
+  const [title] = rawTitle.indexOf(EXECUTING)
+    ? rawTitle.split(EXECUTING)
+    : [title]
+  const quorumProgress = getQuorumProgress(vote)
+  const totalVotes = yea + nay
+  const votesYeaVotersSize = safeDiv(yea, totalVotes)
+  const votesNayVotersSize = safeDiv(nay, totalVotes)
+  const youVoted =
+    connectedAccountVote === VOTE_YEA || connectedAccountVote === VOTE_NAY
+  const handleVoteNo = useCallback(() => {
+    onVote(voteId, VOTE_NAY)
+  }, [onVote, voteId])
+  const handleVoteYes = useCallback(() => {
+    onVote(voteId, VOTE_YEA)
+  }, [onVote, voteId])
+
+  const handleExecute = useCallback(() => {
+    onExecute(voteId)
+  }, [onExecute, voteId])
+  const [yeaPct, nayPct] = percentageList(
+    [votesYeaVotersSize, votesNayVotersSize],
+    2
+  )
+
+  if (!vote) {
+    return null
+  }
+
+  return (
+    <Split
+      primary={
+        <Box>
+          <div
+            css={`
+              display: flex;
+              justify-content: space-between;
+            `}
+          >
+            <Tag mode="app">App Badge</Tag>
+            {youVoted && (
+              <div
+                css={`
+                  display: inline-grid;
+                  grid-template-columns: auto auto;
+                  grid-gap: ${0.5 * GU}px;
+                  align-items: center;
+                  justify-content: center;
+                  height: 20px;
+                  width: auto;
+                  border-radius: 100px;
+                  padding: 0 ${1 * GU}px;
+                  background: ${theme.infoSurface.alpha(0.08)};
+                  color: ${theme.info};
+                  ${textStyle('label2')};
+                `}
+              >
+                <IconCheck size="tiny" /> Voted
+              </div>
+            )}
+          </div>
+          <div
+            css={`
+              display: grid;
+              grid-template-columns: auto;
+              grid-gap: ${2.5 * GU}px;
+              margin-top: ${2.5 * GU}px;
+            `}
+          >
+            <div
+              css={`
+                ${textStyle('title2')};
+              `}
+            >
+              <span css="font-weight: bold;">Vote #{voteId}:</span>{' '}
+              <VoteText text={title} />
```

Comment:
Ok, that link does not really resolve to a specific component, can you tag me in the actual file and I'll review there?

Reply:
Hopefully this link will work: https://www.figma.com/file/O09HJRCFDwDTni4VE3tiRvwI/Voting?node-id=1604%3A1442

Otherwise I've also left a comment there :).

Reply:
55eead6 applies the new designs:

- [Card](https://www.figma.com/file/O09HJRCFDwDTni4VE3tiRvwI/Voting?node-id=42%3A218)
- [Detail](https://www.figma.com/file/O09HJRCFDwDTni4VE3tiRvwI/Voting?node-id=42%3A218)